modules/coredns/collect.go

```diff
@@ -37,6 +51,20 @@ func (cd *CoreDNS) collect() (map[string]int64, error) {
 	}
 
 	mx := newMetrics()
+	if cd.version == nil {
+		cd.version = cd.Version(raw)
+		if cd.version.LTE(version169) {
```

Comment:
> if cd.version.LTE(version169)

How does it work if `cd.Version(raw)` returns nil?

Reply:
Atm the moment the collected metric keys will be empty strings which will not provide us any data. We could set a default version but that's also not guaranteed to work if version fails or we could skip the collection of these metrics if we can't find the version. I think the seconds solution is better. What do you think?

Reply:
What I mean is:
 - "cd.Version" returns nil on error.
 - it means that "if cd.version.LTE(version169)" will panic

Reply:
oh, I see the issue now. I've pushed a fix along with some small refactoring and a unit test to cover the case where the server doesn't report the version. In this case from now on we will not try to collected the 4 metrics that require the server version.